llvm/tools/llvm-c-test/echo.cpp

```diff
@@ -754,7 +755,13 @@ struct FunCloner {
         LLVMSetAlignment(Dst, LLVMGetAlignment(Src));
         LLVMSetOrdering(Dst, LLVMGetOrdering(Src));
         LLVMSetVolatile(Dst, LLVMGetVolatile(Src));
-        LLVMSetAtomicSingleThread(Dst, LLVMIsAtomicSingleThread(Src));
+        if (LLVMIsAtomic(Src)) {
+          LLVMBool IsAtomicSingleThread = LLVMIsAtomicSingleThread(Src);
+          if (IsAtomicSingleThread)
+            LLVMSetAtomicSingleThread(Dst, IsAtomicSingleThread);
+          else
+            LLVMSetAtomicSyncScopeID(Dst, LLVMGetAtomicSyncScopeID(Src));
```

Comment:
Just this line should be sufficient? The single thread case is a subset.

If the intention here is to retain some test coverage for LLVMIsAtomicSingleThread, I don't think echo.cpp is the place for that. You could do that in a unit test.

(Also applies to the changes below.)

Reply:
Yeah, the intention was test coverage. I'll move it to a unit test.